File: PYTHON CLASS/isort.py
def iSort(arr):
    for i in range(1, len(arr)):
 
        key = arr[i]
        ptr = i-1
        
        while j >= 0 and key < arr[ptr] :
                arr[ptr + 1] = arr[ptr]
                ptr -= 1
        arr[ptr + 1] = key
print("Enter the array elements:")        
arr = list(map(int,input().split()))
iSort(arr)
print("Elements after sorting:")
for i in range(len(arr)):
    print ("% d" % arr[i])

File: MY C PROGRAMS/isort.c
//PROGRAM TO ILLUSTRATE INSERTION SORT.

#include<stdio.h>

void main()
{
	int a[10],n,ptr,temp,i;
	printf("Enter the number of elements:");
	scanf("%d",&n);
	//filling array
	for(i=0;i<n;i++)
	{
		printf("Enter element for a[%d]",i);
		scanf("%d",&a[i]);
	}
	printf("\n----Unsorted array-------\n");
	for(i=0;i<n;i++)
	{
		printf("%d\t",a[i]);
	}
	//sorting using insertion sort
	for(i=1;i<n;i++)
	{
		temp=a[i],ptr=i-1;
		while(a[ptr]>temp&&ptr>=0)
		{
			a[ptr+1]=a[ptr];
			ptr--;
		}
		a[ptr+1]=temp;
	}
	printf("\n----Sorted array-------\n");
	for(i=0;i<n;i++)
	{
		printf("%d\t",a[i]);
	}
	
}
